locations/storefinders/storemapper.py

```diff
@@ -21,3 +22,26 @@ def parse(self, response, **kwargs):
 
     def parse_item(self, item, location, **kwargs):
         yield item
+
+    @staticmethod
+    def storefinder_exists(response: Response) -> bool:
+        # Example: https://francescas.com/store-locator
+        if len(response.xpath("//script[@data-storemapper-id]/@data-storemapper-id")) > 0:
+            return True
+
+        if len(response.xpath('//script[contains(text(), "https://www.storemapper.co/js/widget-3.min.js")]')) > 0:
+            return True
+
+        if len(response.xpath('//script[contains(@src, "https://www.storemapper.co/js/widget-3.min.js")]')) > 0:
+            return True
+
+        if len(response.xpath('//div[id="storemapper"]')) > 0:
+            return True
+
+        return False
+
+    @staticmethod
+    def extract_spider_attributes(response: Response) -> dict:
+        return {
+            "key": response.xpath("//script[@data-storemapper-id]/@data-storemapper-id").get(),
```

Comment:
Simpler XPath of `//script/@data-storemapper-id` has the same effect here and in `storefinder_exists`.

Otherwise looks good.